### PR TITLE
pkg/manifest: Add methods to support capability implicit enablement

### PR DESCRIPTION
**Pull request by jottofar**
to check if manifest is enabled and to return manifest's capabilities.

### CODE REVIEW

File: pkg/manifest/manifest.go [hunk 1]
```diff
@@ -60,6 +61,10 @@ type Manifest struct {
 	Obj *unstructured.Unstructured
 }
 
+func (r *resourceId) Equal(id *resourceId) bool {
```
Comment (wking) [inline pkg/manifest/manifest.go:64]: I have mixed feelings about a public method on a private type.  If this needs to be public, can we make `ResourceID` public?

File: pkg/manifest/manifest.go [hunk 2]
```diff
@@ -151,17 +156,78 @@ func (m *Manifest) Include(excludeIdentifier *string, includeTechPreview *bool,
 	return nil
 }
 
+// ManifestEnabled returns whether the manifest is enabled given the defined cluster capabilities.
+func (m *Manifest) ManifestEnabled(capabilities configv1.ClusterVersionCapabilitiesStatus) bool {
```
Comment (LalatenduMohanty) [inline pkg/manifest/manifest.go:160]: s/ManifestEnabled/IsManifestEnabled/

File: pkg/manifest/manifest.go [hunk 3]
```diff
@@ -151,17 +156,78 @@ func (m *Manifest) Include(excludeIdentifier *string, includeTechPreview *bool,
...
+	return enabled
+}
+
+// GetManifestCapabilities returns the manifest's capabilities.
+func (m *Manifest) GetManifestCapabilities() []configv1.ClusterVersionCapability {
+	annotations := m.Obj.GetAnnotations()
+	if annotations == nil {
+		return nil
+	}
+	return getManifestCapabilities(annotations)
```
Comment (LalatenduMohanty) [inline pkg/manifest/manifest.go:183]: You can write 
```
if annonations != nil {
   return getManifestCapabilities(annotations)
}

File: pkg/manifest/manifest.go [hunk 4]
```diff
@@ -151,17 +156,78 @@ func (m *Manifest) Include(excludeIdentifier *string, includeTechPreview *bool,
...
+		allCaps[i] = configv1.ClusterVersionCapability(c)
+	}
+	return allCaps
+}
+
+func getCapabilityProperties(capability configv1.ClusterVersionCapability,
+	capabilities configv1.ClusterVersionCapabilitiesStatus) (isKnown, isEnabled bool) {
+
+	isKnown = false
+	isEnabled = false
```
Comment (LalatenduMohanty) [inline pkg/manifest/manifest.go:204]: default value of a bool variable is "false" , I wonder if we really need to assign false to them

File: pkg/manifest/manifest.go [hunk 5]
```diff
@@ -151,17 +156,78 @@ func (m *Manifest) Include(excludeIdentifier *string, includeTechPreview *bool,
...
+	}
+	for _, enabledCapability := range capabilities.EnabledCapabilities {
+		if capability == enabledCapability {
+			isEnabled = true
+			break
+		}
+
+	}
+	return true, isEnabled
+}
```
Comment (LalatenduMohanty) [inline pkg/manifest/manifest.go:222]: I think we should break this function in to two different functions i.e. one for known capabilities and another for enabled capabilities.

**Review by LalatenduMohanty — CHANGES_REQUESTED** [review]

File: pkg/manifest/manifest.go (hunk 5 above)
Comment (LalatenduMohanty) [reply pkg/manifest/manifest.go:222]: That way it will be easier to maintain

File: pkg/manifest/manifest.go [hunk 6]
```diff
@@ -151,17 +156,78 @@ func (m *Manifest) Include(excludeIdentifier *string, includeTechPreview *bool,
 	return nil
 }
 
+// ManifestEnabled returns whether the manifest is enabled given the defined cluster capabilities.
```
Comment (LalatenduMohanty) [inline pkg/manifest/manifest.go:159]: We should move `func checkResourceEnablement` just after `func (m *Manifest) Include` and the new methods can be inserted after that. As `checkResourceEnablement` only used in the include method. Just a nit pick, not something blocking this PR from merging

File: pkg/manifest/manifest.go (hunk 1 above)
Comment (jottofar) [reply pkg/manifest/manifest.go:64]: I don't see an issue. The private type merely says nothing outside the package can or needs to reference the type. 

I do need to make `Manifest.id` public though.

File: pkg/manifest/manifest.go (hunk 3 above)
Comment (jottofar) [reply pkg/manifest/manifest.go:183]: I can do that but normally lint would bitch about not having a return at the end of a function.

File: pkg/manifest/manifest.go (hunk 4 above)
Comment (jottofar) [reply pkg/manifest/manifest.go:204]: You are correct. Also on reading up a bit on `named parameters` seems more appropriate to have one bare return at the end of the function for readability so I may do that as well.

File: pkg/manifest/manifest.go (hunk 5 above)
Comment (jottofar) [reply pkg/manifest/manifest.go:222]: Yeah, I agree. But this method can just go away if I get rid of `ManifestEnabled`.

File: pkg/manifest/manifest.go (hunk 2 above)
Comment (jottofar) [reply pkg/manifest/manifest.go]: method removed

**Review by LalatenduMohanty — APPROVED** [review]
/lgtm

**Pull request merged**